### PR TITLE
xontrib load/unload

### PR DESCRIPTION
**Pull request by jnoortheen**
- [x] implement unloading

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: docs/tutorial_xontrib.rst [hunk 1]
```diff
@@ -24,15 +24,50 @@ took inspiration from for xonsh:
...
 
-
 Structure
-==========
-Xontribs are modules written in either xonsh (``*.xsh``) or Python (``*.py``).
+================
+Xontribs are modules with some special functions written
+in either xonsh (``*.xsh``) or Python (``*.py``)  that has a couple of special functions to load and unload it.
+
+It is inspired from
```
Comment (yaxollum) [inline docs/tutorial_xontrib.rst:39]: @jnoortheen Did you accidentally delete the rest of this sentence?

Comment (jnoortheen) [reply docs/tutorial_xontrib.rst:39]: I think that is the case, there should be ipython,sphinx ... in the mix

Comment (jnoortheen) [reply docs/tutorial_xontrib.rst:39]: Ok they are above, I should be removing the hanging sentence

**Comment by anki-code** [comment]
How to create autoloadable xontrib like mentioned in #3275?

**Comment by jnoortheen** [comment]
@anki-code it is not implemented yet. I am planning to do so after this PR. basically we will autoload xontribs with entrypoints, there will be option to control this behaviour .

**Comment by anki-code** [comment]
Cool! Two points around this JFYI:
* probably there is some management of loading priority is needed
* there is `events.on_post_init` event ([example](https://github.com/xonsh/xonsh/issues/3582#issuecomment-632688394)) and probably it will be cool to have something like `__rc_xontrib_post_init` too

**Review by anki-code — APPROVED** [review]

**Comment by jnoortheen** [comment]
- Yes I thought about it too. We first implement a naive version and improve upon it, besides the xontribs can utilize events themselves. xontribs should be loaded at `on_pre_rc ` so the rc file can override any environment variables it sets.
- yes it would have a specific event for it to track the time as well

**Comment by jnoortheen** [comment]
Thanks @yaxollum and @anki-code for the reviews

**Pull request merged**